plugins/connectors/mongodb/config.go

```diff
@@ -0,0 +1,141 @@
+/* Copyright © INFINI LTD. All rights reserved.
+ * Web: https://infinilabs.com
+ * Email: hello#infini.ltd */
+
+package mongodb
+
+import (
+	"fmt"
+	"time"
+)
+
+// Config defines the configuration for the MongoDB connector
+type Config struct {
+	// Connection configuration
+	ConnectionURI string `config:"connection_uri"`
+	Database      string `config:"database"`
+	AuthDatabase  string `config:"auth_database"` // Authentication database (e.g., "admin")
+	ClusterType   string `config:"cluster_type"`   // Cluster type: "standalone", "replica_set", "sharded"
+
+	// Collections configuration
+	Collections []CollectionConfig `config:"collections"`
+
+	// Pagination configuration
+	Pagination bool `config:"pagination"`
+	PageSize   int  `config:"page_size"`
+
+	// Last modified field for incremental sync
+	LastModifiedField string `config:"last_modified_field"`
+
+	// Performance optimization configuration
+	BatchSize   int    `config:"batch_size"`
+	Timeout     string `config:"timeout"`
+	MaxPoolSize int    `config:"max_pool_size"`
+
+	// Sync strategy
+	SyncStrategy string `config:"sync_strategy"`
+
+	// Field mapping configuration
+	FieldMapping *FieldMappingConfig `config:"field_mapping"`
+
+	// Advanced query optimization
+	EnableProjection bool `config:"enable_projection"` // Enable projection pushdown
+	EnableIndexHint  bool `config:"enable_index_hint"` // Enable index hints for better performance
+}
+
+type CollectionConfig struct {
+	Name           string                 `config:"name"`
+	Filter         map[string]interface{} `config:"filter"`
+	TitleField     string                 `config:"title_field"`
```

Comment:
What are these configurations used for? The field mapping should already determine the field selection.